File: PyTAT/tests/test_split_to_no_edge.py
# -*- coding: utf-8 -*-
import TAT


def test_basic_usage_0():
    a = TAT.Z2.S.Tensor(["i", "j"], [[(False, 2)], [(False, 1)]]).range_()
    b = a.split_edge({"i": [("k", [(False, 2)])], "j": []})


def test_basic_usage_1():
    a = TAT.Z2.S.Tensor(["i", "j"], [[(False, 2), (True, 2)], [(False, 1)]]).range_()
    b = a.split_edge({"i": [("k", [(False, 2), (True, 2)])], "j": []})

File: TAT/tests/test_split_to_no_edge.cpp
#include <TAT/TAT.hpp>
#include <gtest/gtest.h>

TEST(test_split_to_no_edge, basic_usage_0) {
    auto a = TAT::Tensor<float, TAT::BoseZ2Symmetry>({"i", "j"}, {{{false, 2}}, {{false, 1}}}).range_();
    auto b = a.split_edge({
        {"i", {{"k", {std::vector<std::pair<TAT::BoseZ2Symmetry, TAT::Size>>{{false, 2}}}}}},
        {"j", {}},
    });
}

TEST(test_split_to_no_edge, basic_usage_1) {
    auto a = TAT::Tensor<float, TAT::BoseZ2Symmetry>({"i", "j"}, {{{false, 2}, {true, 2}}, {{false, 1}}}).range_();
    auto b = a.split_edge({
        {"i", {{"k", {std::vector<std::pair<TAT::BoseZ2Symmetry, TAT::Size>>{{false, 2}, {true, 2}}}}}},
        {"j", {}},
    });
}
